File: scripts/info.lua
function uftrace_begin(ctx)
    print(ctx['record'])
    print(ctx['version'])
    io.write('(')
    for _, cmd in ipairs(ctx['cmds']) do
        io.write("'" .. cmd .. "', ")
    end
    print(')')
end

function uftrace_entry(ctx)
end

function uftrace_exit(ctx)
end

File: scripts/info.c
#include <stdbool.h>
#include <stdint.h>
#include <stdio.h>
#include <uftrace/script.h>

void uftrace_begin(struct uftrace_script_info *sc_info)
{
	printf("%d\n", sc_info->record);
	printf("%s\n", sc_info->version);
	printf("%s\n", sc_info->name);
	printf("%s\n", sc_info->cmds);

	/* disable actual data record by uftrace */
	sc_info->record = false;
}

void uftrace_entry(struct uftrace_script_base_ctx *sc_ctx)
{
}

void uftrace_exit(struct uftrace_script_base_ctx *sc_ctx)
{
}

void uftrace_end(void)
{
}
